lib/diagnostic.js

```diff
@@ -7,6 +7,20 @@
 // `let minimum = array.reduce(min);`
 const min = function (previous, current) {
 
+  // use an accumulator
+  for (let i = 0; i < previous.length; i++) {
+
+    // create empty array
+    let minValue = []
+
+    // this assigns the previous array elements to the minValue
+    // array so that the original array is not modified
+    minValue[i] = previous[i]
+
+    // use the Math.min function to find the min value in an array.
+    current = Math.min(minValue)
```

Comment:
I don't believe you can actually use `Math.min` on an array, you would have to pass it individual numbers as arguments.